forum/qa-content/qa-page.js

```diff
@@ -186,14 +186,232 @@ function qa_ajax_error()
  */
 ;(function(document)
 {
+	////////////
+	// clear console (for devloping purposes - clears errors from console)
+	////console.clear();
+	////////////
+	
 	'use strict';
 	
+	// boolean flag, to check if browser supports 'select()' and 'copy' command
+	var prepareClipboardCopying;
+	
+	if (window.getSelection && document.queryCommandSupported('copy'))
+		prepareClipboardCopying = true;
+	else 
+		prepareClipboardCopying = false;
+	
+	
+	/*
+	 * Feature: preview HTML/CSS/JavaScript code from chosen post in codepen.io / jsfiddle.net
+	 */
+	 function viewHtmlCssJs()
+	 {
+		﻿
+		/*
+		 * Adapted code from Codepen API tutorial: https://blog.codepen.io/documentation/api/prefill/
+		 */
+		function codepenSnippet(codepenData, parent)
+		{
+			var JSONstring = JSON.stringify(codepenData)
+				// Quotes will screw up the JSON
+				.replace(/"/g, "&​quot;") // careful copy and pasting, I had to use a zero-width space here to get markdown to post this.
+				.replace(/'/g, "&apos;");
+
+			var form = document.createElement('form');
+			form.action = 'http://codepen.io/pen/define';
+			form.method = 'POST';
+			form.target= '_blank';
+			form.classList.add('codepen-snippet');
+
+			var input1 = document.createElement('input');
+			input1.type = 'hidden';
+			input1.name = 'data';
+			input1.value = JSONstring;
+
+			var input2 = document.createElement('input');
+			input2.type = 'image';
+			input2.src = 'https://assets.codepen.io/assets/logos/codepen-logo-9c0933e8569e634b75ac2eb808da908d.svg';
+			input2.value = 'Create new CODEPEN';

```

Comment:
https://php-kurs.gitbooks.io/phpkurs/content/czesc-i/Zmienne/Dobre-praktyki.html may help you :)